### PR TITLE
Feature/loan list v2

### PR DESCRIPTION
**Pull request by turegjorup**
#### Link to issue

https://platform.dandigbib.org/issues/5387

#### Description

Replaces https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/131

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -72,8 +72,6 @@ services:
       lagoon.type: cli-persistent
       lagoon.persistent.name: nginx # mount the persistent storage of nginx into this container
       lagoon.persistent: /app/web/sites/default/files/ # location where the persistent storage should be mounted
-    environment:
-      << : *default-environment # loads the defined environment variables from the top
```
Comment (cableman) [inline docker-compose.yml:76]: Why this  change to docker setup?

File: web/modules/custom/dpl_loans/src/Controller/DplLoansController.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    /** @var \Drupal\dpl_loans\Plugin\Block\LoanListBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_loans_list_block', $config);
+
+    // Some blocks might implement access check.
+    $access_result = $plugin_block->access($this->currentUser());
+
+    // Return empty render array if user doesn't have access.
+    // $access_result can be boolean or an AccessResult class.
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      // You might need to add some cache tags/contexts.
```
Comment (cableman) [inline web/modules/custom/dpl_loans/src/Controller/DplLoansController.php:65]: Maybe we should lock into ensuring the block is cached? Or remove update the comments in this function.

File: web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    $data = [
+      // Config.
+      "fbs-base-url" => $fbsConfig->get('base_url'),
+      "publizon-base-url" => $publizonConfig->get('base_url'),
+      // This overrides config.
+      "page-size-desktop" => "25",
+      // This overrides config.
+      "page-size-mobile" => "25",
+      // Urls.
+      // @todo update placeholder URL's
```
Comment (cableman) [inline web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:80]: Why this placeholder to images.... why are feed-page-url an image?

File: web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    $context = ['context' => 'Loan list'];
+    $contextAria = ['context' => 'Loan list (Aria)'];
+
+    $fbsConfig = $this->configFactory->get('dpl_fbs.settings');
+    $publizonConfig = $this->configFactory->get('dpl_publizon.settings');
+
+    $data = [
+      // Config.
+      "fbs-base-url" => $fbsConfig->get('base_url'),
+      "publizon-base-url" => $publizonConfig->get('base_url'),
```
Comment (cableman) [inline web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:74]: Should we refactor this to use the same syntax as `DplReactAppsController::searchResultUrl()`?

**Review by cableman — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (turegjorup) [reply docker-compose.yml:76]: Lines were duplicated. See commmit https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/186/commits/c1f62c1e0f62620309deedaf33d6793b58d7cff1

File: web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php (hunk 4 above)
Comment (turegjorup) [reply web/modules/custom/dpl_loans/src/Plugin/Block/LoanListBlock.php:74]: Yes/No. The URL's we get here are external URLs from config. The URL they get in `DplReactAppsController::searchResultUrl()` are internal URLs from routes son not directly comparable.

I think we should have an established way of getting URLs across module boundaries. You could argue that `dpl_loans`should not call an internal(?) function in `dpl_react_apps`.

This should be handled on an architectural level and as such out of scope of this PR.

**Pull request closed without merging**